react-toggle-switch-component/src/toggle-switch.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+
+class ToggleButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = {
+      isOn: false,
```

Comment:
I would recommend removing everything from state except this property, and store/update the other values as properties in the constructor. You always want to keep state as trim as possible. that does not mean you can't have multiple properties in state, just that you don't want to store anything there which does not need to be there.